### PR TITLE
Feat/push api

### PR DESCRIPTION
**Pull request by olastor**
Implementing Push Notifications for Firefox and Chrome

@binarious Should I try implementing Safari as well?

### CODE REVIEW

File: src/app/helper/youtube.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ export default {
     return new Promise((resolve, reject) => {
       Youtube.authenticate({
         type: 'key',
-        key: process.env.YOUTUBE_API_KEY
```
Comment (binarious) [inline src/app/helper/youtube.js:8]: Please rename this in the README, too.

File: src/app/routes/_message.router.js [hunk 2]
```diff
@@ -2,6 +2,7 @@ import Message from '../models/message.model.js';
 let ObjectId = require('mongoose').Types.ObjectId;
 import moment from 'moment';
 
+
```
Comment (binarious) [inline src/app/routes/_message.router.js]: Why?

File: src/app/routes/_push.router.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      const endpoint = req.body.endpoint ? req.body.endpoint : null;
+      const p256dh = req.body.keys && req.body.keys['p256dh'] ? req.body.keys['p256dh'] : '';
+      const auth = req.body.keys && req.body.keys['auth'] ? req.body.keys['auth'] : '';
+
+      if (!endpoint) {
+        res.status(400).send();
+      }
+
+      // Check for already existing subscription
+      const subscription = await PushSubscriptions.findOne({
```
Comment (binarious) [inline src/app/routes/_push.router.js:31]: Isn't there a subscription for every user? Why `findOne` then?

**Review by binarious — CHANGES_REQUESTED** [review]
Great work! Some minor things though.

File: src/app/routes/_push.router.js (hunk 3 above)
Comment (olastor) [reply src/app/routes/_push.router.js:31]: The endpoint url is unique for every session, if that's what you mean.

**Review by binarious — APPROVED** [review]

**Pull request merged**